tests/cache/tests.py

```diff
@@ -1165,6 +1165,16 @@ def test_get_or_set_racing(self):
             cache_add.return_value = False
             self.assertEqual(cache.get_or_set("key", "default"), "default")
 
+    def test_async_impl(self):
```

Comment:
I would suggest a more verbose test method name. You could do away with the comment in that case.